File: Data Structures/queue_using_stacks.py
class Stack:
     def __init__(self):
         self.items = []

     def isEmpty(self):
         return self.items == []

     def push(self, item):
         self.items.append(item)

     def pop(self):
         return self.items.pop()

     def peek(self):
         return self.items[len(self.items)-1]

     def size(self):
         return len(self.items)

class MyQueue(object):
    newest = Stack()
    oldest = Stack()
    def __init__(self):
        self.newest = Stack()
        self.oldest = Stack()
        
    def shiftStacks(self):
        if self.oldest.isEmpty():
            while not self.newest.isEmpty():
                self.oldest.push(self.newest.pop())
    
    def peek(self):
        self.shiftStacks()
        return self.oldest.peek()
        
    def pop(self):
        self.shiftStacks()
        return self.oldest.pop()
        
    def put(self, value):
        self.newest.push(value)

queue = MyQueue()
t = int(input())
for line in range(t):
    values = map(int, input().split())
    values = list(values)
    if values[0] == 1:
        queue.put(values[1])        
    elif values[0] == 2:
        queue.pop()
    else:
        print(queue.peek())
        

File: Data Structures/queue_using_stacks.java
import java.io.*;
import java.util.*;
import java.text.*;
import java.math.*;
import java.util.regex.*;

class MyQueue<Integer>{
    Stack<Integer> oldest = new Stack<Integer>();
    Stack<Integer> newest = new Stack<Integer>();
    
    public void shiftStacks(){
        if (this.oldest.isEmpty()){
            while (!this.newest.isEmpty()){
                this.oldest.push(this.newest.pop());
            }
        }
    }
    
    public void enqueue(Integer item){
        this.newest.push(item);
    }
    
    public Integer dequeue(){
        shiftStacks();
        return this.oldest.pop();        
    }
    
    public Integer peek(){
        shiftStacks();
        return this.oldest.peek();     
    }
    
    public boolean isEmpty(){
        return (this.oldest == null & this.newest == null);
    }
    
}

public class Solution {
    public static void main(String[] args) {
        MyQueue<Integer> queue = new MyQueue<Integer>();

        Scanner scan = new Scanner(System.in);
        int n = scan.nextInt();

        for (int i = 0; i < n; i++) {
            int operation = scan.nextInt();
            if (operation == 1) { // enqueue
              queue.enqueue(scan.nextInt());
            } else if (operation == 2) { // dequeue
              queue.dequeue();
            } else if (operation == 3) { // print/peek
              System.out.println(queue.peek());
            }
        }
        scan.close();
    }
}
